processor/spanmetricsprocessor/processor.go

```diff
@@ -364,7 +368,8 @@ func (p *processorImp) getDimensionsByMetricKey(k metricKey) (pcommon.Map, error
 // Each metric is identified by a key that is built from the service name
 // and span metadata such as operation, kind, status_code and any additional
 // dimensions the user has configured.
-func (p *processorImp) aggregateMetrics(traces ptrace.Traces) {
+func (p *processorImp) aggregateMetrics(traces ptrace.Traces) []metricKey {
+	var metricKeys []metricKey
```

Comment:
Nice, I prefer this approach of the processor not maintaining the `metricKey` state.

Reply:
:-)